### PR TITLE
[Merged by Bors] - Add EdSignature type for ed25519 Signatures

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
Part of #4139: replace `[]byte` for ed25519 signatures with a new custom type `EdSignature` (`[32]byte`)

## Changes
- new type `types.EdSignature` with some additional methods and helpers for tests
  - replace all existing occurrences of `[]byte` for ed25519 signatures that are scale encoded with `types.EdSignature`
  - pull up the changes the call chain, replace `string` and `[]byte` with new type
- Update existing code and tests
- In some places where PublicKey was used replace it by NodeID
  - its the same data anyway but calling `PublicKey` requires at least 1 allocation, calling `NodeID` is generally free.

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: common/types/signatures.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	return scale.DecodeByteArray(decoder, s[:])
+}
+
+// String returns a string representation of the NodeID, for logging purposes.
+// It implements the Stringer interface.
+func (s *EdSignature) String() string {
+	return hex.EncodeToString(s.Bytes())
+}
+
+// Bytes returns the byte representation of the Edwards public key.
```
Comment (countvonzero) [inline common/types/signatures.go]: :s/public key/signature

File: beacon/handlers.go [hunk 2]
```diff
@@ -214,8 +214,7 @@ func (pd *ProtocolDriver) verifyProposalMessage(logger log.Log, m ProposalMessag
 		return fmt.Errorf("[proposal] verify VRF (miner ID %v): %w", minerID, errVRFNotVerified)
```
Comment (countvonzero) [inline beacon/handlers.go]: might as well rename all the minerID to nodeID

**Review by countvonzero — APPROVED** [review]

**Comment by countvonzero** [comment]
> Merging [#4210] will decrease coverage by 0.1%.

lol. this is terrible!

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**